external-crates/move/crates/move-analyzer/src/symbols.rs

```diff
@@ -1427,17 +1434,45 @@ impl SymbolicatorRunner {
                                 }
                                 continue;
                             }
-                            pkgs_to_analyze
-                                .entry(root_dir.unwrap())
-                                .or_insert_with(BTreeSet::new)
-                                .insert(starting_path.clone());
+                            // we want to collect either all modified Move files (as BTreeSet<Some(PathBuf)> or,
+                            // if one of the paths for a package is not a Move file, have aonly one BTreeSet
+                            // None entry. This will be later passed to symbolicator as Some(Vec<PathBuf>) or
+                            // None, to indicate if incremental symbolication is possible or not.
+                            let root_dir = root_dir_opt.unwrap();
+                            if let Some(mut modified_files) = pkgs_to_analyze.remove(&root_dir) {
+                                // we already seen this package
+                                if modified_files.len() != 1 || modified_files.first().is_some() {
```

Comment:
What is this testing for and why not just `!modified_files.is_empty()`?

Reply:
It could be `!modified_files.is_empty()`. The intention was to represent a situation when `modified_files` has only one entry and this entry is `None`, but since an entry cannot be `None` if other entries exist, checking of non-emptiness is totally fine as well